stdlib/test/builtin/test_string.mojo

```diff
@@ -1082,6 +1082,73 @@ def test_indexing():
     assert_equal(a[2], "c")
 
 
+def test_format_args():
+    assert_equal(String("{0} {1}").format("Hello", "world"), "Hello world")
+
+    assert_equal(String("Number: {0}").format(123), "Number: 123")
+
+    assert_equal(
+        String("{0} is fun and fast {1}").format("Mojo", "🔥"),
+        "Mojo is fun and fast 🔥",
+    )
+    assert_equal(String("{{0}} {{{1}}}").format("123", 456), "{123} {{456}}")
+    assert_equal(String("{{0}} {{{1}}}").format(), "{{0}} {{{1}}}")
+
+    assert_equal(String("").format(), "")
+
+    assert_equal(String("Is {0} even? {1}").format(2, True), "Is 2 even? True")
+
+    assert_equal(String("{0} {1}").format("{0}", True), "{0} True")
+
+    with assert_raises(
+        contains="Recursion with a replacement value containing bracket"
+    ):
+        print(String("Is {0} even? {1}").format("{1}", True))
+
+    with assert_raises(contains="args[1] was provided but not used"):
```

Comment:
Whats the behaviour if the number of args is less expected? That should raise as well.